test/monitoring_console/monitoring_console_test.go

```diff
@@ -0,0 +1,513 @@
+				return count
+			}, deployment.GetTimeout(), PollInterval).Should(Equal(1))
+
+			// Monitoring Console Pod is in Ready State
+			Eventually(func() bool {
+				check := testenv.CheckMCPodReady(testenvInstance.GetName())
+				return check
+			}, deployment.GetTimeout(), PollInterval).Should(Equal(true))
+
+			// Only 2 peer expected in MC peer list
+			peerList = testenv.GetConfiguredPeers(testenvInstance.GetName())
+			testenvInstance.Log.Info("Peers in configuredPeer List", "count", len(peerList))
+			Expect(len(peerList)).To(Equal(2))
+
+			// Check standalone pods are configured  in MC Peer List
+			found := make(map[string]bool)
+			testenvInstance.Log.Info("Peer List", "instance", peerList)
+			for i := 0; i < 2; i++ {
+				podName := "splunk-" + deployment.GetName() + "-standalone-" + strconv.Itoa(i)
+				found[podName] = false
+				for _, peer := range peerList {
+					testenvInstance.Log.Info("Check Peer matches standalone pod", "Standalone Pod", podName, "Peer in peer list", peer)
+					if strings.Contains(peer, podName) {
+						found[podName] = true
+						break
+					}
+				}
+			}
+			allStandaloneConfigured := true
+			for _, key := range found {
+				if !key {
+					allStandaloneConfigured = false
+					break
+				}
+			}
+			Expect(allStandaloneConfigured).To(Equal(true))
+		})
+	})
+
+	Context("SearchHeadCluster deployment with Scale Up", func() {
+		It("MC can configure SHC instances in a namespace", func() {
+
+			_, err := deployment.DeploySearchHeadCluster(deployment.GetName(), "", "", "")
+			Expect(err).To(Succeed(), "Unable to deploy search head cluster")
+			// Ensure search head cluster go to Ready phase
+			shc := &enterprisev1.SearchHeadCluster{}
+			Eventually(func() splcommon.Phase {
+				err := deployment.GetInstance(deployment.GetName(), shc)
+				if err != nil {
+					return splcommon.PhaseError
+				}
+				testenvInstance.Log.Info("Waiting for search head cluster instance status to be ready", "instance", shc.ObjectMeta.Name, "Phase", shc.Status.Phase)
+				dumpGetPods(testenvInstance.GetName())
+				return shc.Status.Phase
+			}, deployment.GetTimeout(), PollInterval).Should(Equal(splcommon.PhaseReady))
+
+			// In a steady state, we should stay in Ready and not flip-flop around
+			Consistently(func() splcommon.Phase {
+				_ = deployment.GetInstance(deployment.GetName(), shc)
+				return shc.Status.Phase
+			}, ConsistentDuration, ConsistentPollInterval).Should(Equal(splcommon.PhaseReady))
+
+			// Check Monitoring Console Pod is in Ready State
+			Eventually(func() bool {
+				check := testenv.CheckMCPodReady(testenvInstance.GetName())
+				return check
+			}, deployment.GetTimeout(), PollInterval).Should(Equal(true))
+
+			// Check Monitoring console is configured with all search head instances in namespace
+			peerList := testenv.GetConfiguredPeers(testenvInstance.GetName())
+			found := make(map[string]bool)
+			testenvInstance.Log.Info("Peer List", "instance", peerList)
+			for i := 0; i < 3; i++ {
```

Comment:
Nit: Just curious how we are getting number '3' here. It will be nice if we can refer one common variable

Reply:
We create 3 searchheads by default. In this case while deploying SHC we are not specifying any replicas during creation of SHC. Although it makes sense to have a variable for clarity.

Reply:
+1, agree with Subba. Something like "MinimumNodesCluster" which can apply for IDXC and SHC